src/test/java/com/artipie/vertx/VertxSliceServerTest.java

```diff
@@ -23,76 +23,182 @@
-            ),
-            port
-        )) {
-            server.start();
-            final WebClient web = WebClient.create(vertx);
-            final String actual = web.get(port, VertxSliceServerTest.HOST, uri)
-                .rxSend()
-                .blockingGet()
-                .bodyAsString();
-            MatcherAssert.assertThat(actual, Matchers.equalTo(expected));
-            web.close();
-        }
+            )
+        );
+        final String actual = this.client.get(this.port, VertxSliceServerTest.HOST, "/hello1")
+            .rxSend()
+            .blockingGet()
+            .bodyAsString();
+        MatcherAssert.assertThat(actual, Matchers.equalTo(expected));
+    }
+
+    @Test
+    public void exceptionInSlice() {
+        this.start(
+            (line, headers, body) -> {
+                throw new IllegalStateException("Failed to create response");
+            }
+        );
+        final int code = this.client.get(this.port, VertxSliceServerTest.HOST, "")
+            .rxSend()
+            .blockingGet()
+            .statusCode();
+        MatcherAssert.assertThat(
+            code,
+            Matchers.equalTo(HttpURLConnection.HTTP_INTERNAL_ERROR)
+        );
+    }
+
+    @Test
+    public void exceptionInResponse() {
+        this.start(
+            (line, headers, body) -> connection -> {
+                throw new IllegalStateException("Failed to send response");
+            }
+        );
+        final int code = this.client.get(this.port, VertxSliceServerTest.HOST, "")
+            .rxSend()
+            .blockingGet()
+            .statusCode();
+        MatcherAssert.assertThat(
+            code,
+            Matchers.equalTo(HttpURLConnection.HTTP_INTERNAL_ERROR)
+        );
+    }
+
+    @Test
+    public void exceptionInResponseAsync() {
+        this.start(
+            (line, headers, body) -> connection -> CompletableFuture.runAsync(
+                () -> {
+                    throw new IllegalStateException("Failed to send response async");
+                }
+            )
+        );
+        final int code = this.client.get(this.port, VertxSliceServerTest.HOST, "")
+            .rxSend()
+            .blockingGet()
+            .statusCode();
+        MatcherAssert.assertThat(
+            code,
+            Matchers.equalTo(HttpURLConnection.HTTP_INTERNAL_ERROR)
+        );
+    }
+
+    @Test
+    public void exceptionInBody() {
+        this.start(
+            (line, headers, body) -> connection -> connection.accept(
+                RsStatus.OK,
+                Collections.emptyList(),
+                Flowable.error(new IllegalStateException("Failed to publish body"))
+            )
+        );
+        final int code = this.client.get(this.port, VertxSliceServerTest.HOST, "")
+            .rxSend()
+            .blockingGet()
+            .statusCode();
+        MatcherAssert.assertThat(
+            code,
```

Comment:
@olegmoz I suggest to modify this test to assert that the body contains the text `IllegalStateException`. This will be more compliant with the test method name.